File: simple/go/part5/internal/interpreteter/token.go
package interpreteter

const (
	INTEGER = "0-9"
	PLUS    = "+"
	MINUS   = "-"
	MUL     = "*"
	DIV     = "/"
	NEWLINE = "\\n"
)

type Token interface {
	Type() string
	Value() int
}

type TokenFactory interface {
	NewToken(value int, kind string) Token
}

File: simple/rust/part7/src/token.rs
#[derive(PartialEq, Debug, Clone, Copy)]
pub enum Type {
    EOF,
    MUL,
    DIV,
    PLUS,
    MINUS,
    INTEGER,
    LPAREN,
    RPAREN,
}

#[derive(Debug, Clone, Copy)]
pub struct Token {
    kind: Type,
    value: i32,
}

impl Token {
    pub fn new(kind: Type, value: i32) -> Token {
        Token {
            kind: kind,
            value: value,
        }
    }

    pub fn kind(&self) -> Type {
        self.kind
    }

    pub fn value(&self) -> i32 {
        self.value
    }
}
